annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+    ) as mock_get_user_names_by_user_ids, patch(
+        "annotation.tasks.services.get_file_names_by_file_ids"
+    ) as mock_get_file_names_by_file_ids, patch(
+        "annotation.tasks.services.io.BytesIO"
+    ) as mock_bytes_io, patch(
+        "annotation.tasks.services.io.TextIOWrapper"
+    ) as mock_text_io_wrapper:
+
+        mock_get_user_names_by_user_ids.return_value = mock_user_name
+        mock_get_file_names_by_file_ids.return_value = mock_file_name
+
+        mock_binary = MagicMock()
+        mock_bytes_io.return_value = mock_binary
+        mock_text_file = MagicMock()
+        mock_text_io_wrapper.return_value = mock_text_file
+
+        filename, _ = create_export_csv(mock_db, schema, "tenant", "token")
+
+        assert filename == expected_filename
+
+
+@pytest.fixture
+def mock_task():
+    task = MagicMock(ManualAnnotationTask)
+    task.id = 1
+    task.user_id = 2
+    task.job_id = 10
+    task.file_id = 3
+    task.pages = {1, 2, 3}
+    task.is_validation = False
+    return task
+
+
+@pytest.fixture
+def mock_tenant_data():
+    mock_data = MagicMock(TenantData)
+    mock_data.token = "mock_token"
+    return mock_data
+
+
+def test_evaluate_agreement_score(
+    mock_session: Mock, mock_task: Mock, mock_tenant_data: Mock
+):
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket"
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_agreement_score"
+    ) as mock_get_agreement_score, patch(
+        "annotation.tasks.services.compare_agreement_scores"
+    ) as mock_compare_agreement_scores:
+
+        mock_get_file_path_and_bucket.return_value = (
+            "mock_s3_file_path",
+            "mock_s3_file_bucket",
+        )
+        mock_agreement_score_response = [
+            MagicMock(AgreementScoreServiceResponse)
+        ]
+        mock_get_agreement_score.return_value = mock_agreement_score_response
+        mock_compare_agreement_scores.return_value = MagicMock(
+            AgreementScoreComparingResult
+        )
+
+        mock_query = MagicMock()
+        mock_query.filter.return_value.filter.return_value.all.return_value = [
+            mock_task
+        ]
+        mock_session.query.return_value = mock_query
+
+        result = evaluate_agreement_score(
+            db=mock_session,
+            task=mock_task,
+            tenant="mock_tenant",
+            token=mock_tenant_data,
+        )
+
+        mock_get_file_path_and_bucket.assert_called_once_with(
+            mock_task.file_id, "mock_tenant", mock_tenant_data.token
+        )
+
+        mock_compare_agreement_scores.assert_called_once_with(
+            mock_agreement_score_response, AGREEMENT_SCORE_MIN_MATCH
+        )
+
+        assert isinstance(result, AgreementScoreComparingResult)
+
+
+class ResponseScore:
+    def __init__(self, task_id: int, agreement_score: float):
+        self.task_id = task_id
+        self.agreement_score = agreement_score
```

Comment:
Was there a reason for redefining a ResponseScore class, rather than importing one? Here, now we have two places in our code where we need to refactor it, if we want to make a change